### ISSUE TITLE
Snapshot version: Can't process instrumentationArg [package] (no equal sign)

### ISSUE THREAD

**Issue by bholota** [square/spoon#490]
Issue is occurring when I'm trying to provide instrumentation arg with -e or --es.

Execution:
```
java -jar \
	../../docker/spoon-runner-2.0.0-20171020.002642-36-all.jar \
	--debug \
	App/build/outputs/apk/app-debug-androidTest.apk \
	App/build/outputs/apk/app-debug.apk \
	-e package com.sample.app.ui.testsuites.happypathscenarios.login \
	--output App/build/outputs/spoon
```

Result:
```
2017-10-27 11:26:39 [SDR.createConfiguredRunner] Can't process instrumentationArg [package] (no equal sign)
2017-10-27 11:26:39 [SDR.createConfiguredRunner] Can't process instrumentationArg [com.sample.app.ui.testsuites.happypathscenarios.login] (no equal sign)
```

I did some investigation - issue is connected to different format in `SpoonRunner`. `CliArgs` parses  instrumentation args in '-e KEY VALUE' format, `SpoonDeviceRunner` requires 'key=value' format required by instrumentation. Prolly adding converter in SpoonDeviceRunner (to maintain compatibility with previous versions) would be enough.

**Comment by dweebo** [comment]
According to the instructions in the spoon README you are supposed to pass e arguments as KEY=VALUE.  But yeah that is different behavior than you'd use running android instrumentation directly and probably wouldn't be hard to add.

**Comment by JakeWharton** [comment]
We don't need to maintain compatibility. Let's just fix the runner.

**Comment by bholota** [comment]
@dweebo Current master has different approach (check `CliArgs.kt`), now it's `-e KEY VALUE`. 

@JakeWharton I think I've fixed it. Now parser will be taking optional Map<String, String>. I'll do some cleanup and PR it later. 

Also there is 2nd instrumentation arg related issue: 

After fix, when I passed -e package argument it was parsed properly, runner took proper test method but none was executed. `SpoonDeviceRunner` internally specifies test method to limit test execution, but it cannot limit test execution to package and class at same time. 

It can be worked around - when `SpoonDeviceRunner` has list of active tests package argument can be just removed: `runner.removeInstrumentationArg("package");`
